test/2-nodes.js

```diff
@@ -76,11 +76,16 @@ describe('basics between 2 nodes', () => {
     })
 
     it('Subscribe to a topic:Z in nodeA', (done) => {
+      let subChanges
+      fsB.once('floodsub:subscription-change', (...args) => { subChanges = args })
       fsA.subscribe('Z')
       setTimeout(() => {
         expectSet(fsA.subscriptions, ['Z'])
         expect(fsB.peers.size).to.equal(1)
         expectSet(first(fsB.peers).topics, ['Z'])
+        expect(subChanges[0].id.toB58String()).to.equal(first(fsB.peers).info.id.toB58String())
+        expectSet(subChanges[1], ['Z'])
+        expect(subChanges[2]).to.be.eql([{ topicCID: 'Z', subscribe: true }])
```

Comment:
With this feature, we can make use of it in the tests. This way. we can get rid of flaky tests like this one, which we use a non-deterministic logic (timeout) and can go wrong sometimes.

Example:

```js
fsB.once('floodsub:subscription-change', (...args) => { 
  subChanges = args 

  expectSet(fsA.subscriptions, ['Z'])
  expect(fsB.peers.size).to.equal(1)
  expectSet(first(fsB.peers).topics, ['Z'])
  expect(subChanges[0].id.toB58String()).to.equal(first(fsB.peers).info.id.toB58String())
  expectSet(subChanges[1], ['Z'])
  expect(subChanges[2]).to.be.eql([{ topicCID: 'Z', subscribe: true }])
  done()
})
fsA.subscribe('Z')
```

What do you think?

Reply:
yes, please 😁